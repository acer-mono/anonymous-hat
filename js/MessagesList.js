class MessagesList extends React.Component {
    render() {
        const { messages } = this.props;

        return <>
            { messages.map((message, index) => (
                <Message user={message.user} message={message.message} key={index}/>
                ))}
        </>;
    }
}