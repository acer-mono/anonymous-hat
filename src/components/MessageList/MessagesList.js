import React from 'react';
import Message from '@/components/Message';

function MessagesList({ messages }) {
  return (
    <>
      {messages.map(message => (
        <Message user={message.nickname} message={message.content} key={message.id} />
      ))}
    </>
  );
}

export default MessagesList;
