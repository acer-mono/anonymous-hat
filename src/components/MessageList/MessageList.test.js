import { shallow } from 'enzyme/build';
import MessagesList from '@/components/MessageList';
import React from 'react';
import renderer from 'react-test-renderer';

describe('MessagesList', () => {
  test('add 3 elements', () => {
    const messages = [
      { nickname: 'first', content: 'firstMessage', id: 1 },
      { nickname: 'second', content: 'secondMessage', id: 2 },
      { nickname: 'second', content: 'secondMessage', id: 3 },
    ];
    const component = shallow(<MessagesList messages={messages} />);
    expect(component.find('Message')).toHaveLength(3);
  });

  test('add empty array', () => {
    const messages = [];
    const component = shallow(<MessagesList messages={messages} />);
    expect(component.find('Message')).toHaveLength(0);
  });

  test('MessageList test with snapshot (empty message list)', () => {
    const messages = [];
    const component = renderer.create(<MessagesList messages={messages} />);
    let componentSnapshot = component.toJSON();
    expect(componentSnapshot).toMatchSnapshot();
  });

  test('MessageList test with snapshot (message list with 3 messages)', () => {
    const messages = [
      { nickname: 'first', content: 'firstMessage', id: 1 },
      { nickname: 'second', content: 'secondMessage', id: 2 },
      { nickname: 'second', content: 'secondMessage', id: 3 },
    ];
    const component = renderer.create(<MessagesList messages={messages} />);
    let componentSnapshot = component.toJSON();
    expect(componentSnapshot).toMatchSnapshot();
  });
});
