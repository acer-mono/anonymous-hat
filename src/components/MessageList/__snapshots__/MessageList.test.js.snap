// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessagesList MessageList test with snapshot (empty message list) 1`] = `null`;

exports[`MessagesList MessageList test with snapshot (message list with 3 messages) 1`] = `
Array [
  <div
    className="card mb-3"
  >
    <div
      className="card-header font-weight-bold"
    >
      first
    </div>
    <div
      className="card-body"
    >
      firstMessage
    </div>
  </div>,
  <div
    className="card mb-3"
  >
    <div
      className="card-header font-weight-bold"
    >
      second
    </div>
    <div
      className="card-body"
    >
      secondMessage
    </div>
  </div>,
  <div
    className="card mb-3"
  >
    <div
      className="card-header font-weight-bold"
    >
      second
    </div>
    <div
      className="card-body"
    >
      secondMessage
    </div>
  </div>,
]
`;
